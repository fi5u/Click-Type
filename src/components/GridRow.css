.GridRow {
    border-color: #d2dad1 !important;
    border-radius: 0 !important;
    display: flex;
    margin-bottom: -1px !important;
    opacity: .5;
    overflow: hidden;
    width: 100%;
}

.GridRow--is-active {
    border-color: #aaa !important;
    opacity: 1;
}

.GridRow--is-first {
    border-radius: 0.28571429rem 0.28571429rem 0 0 !important;
}

.GridRow--is-last {
    border-radius: 0 0 0.28571429rem 0.28571429rem !important;
}
