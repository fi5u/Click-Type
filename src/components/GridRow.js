import './GridItem.css'
import './GridRow.css'
import React, {
    Component,
    PropTypes,
} from 'react'
import {
    ButtonGroup,
} from 'react-bootstrap'

class GridRow extends Component {
    render() {
        const {
            children,
            isActive
        } = this.props

        return(
            <ButtonGroup
                className={`GridRow${isActive ? ' GridRow--is-active' : ''}`}
                role="group"
            >
                {children}
            </ButtonGroup>
        )
    }
}

GridRow.propTypes = {
    children: PropTypes.node,
    isActive: PropTypes.bool.isRequired,
}

export default GridRow
