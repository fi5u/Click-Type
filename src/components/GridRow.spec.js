/* global expect, it, jest */
import GridRow from './GridRow'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

const props = {
    activeButtonIteration: 0,
    characters: ['a','b','c'],
    clickButton: jest.fn(),
    isActive: false,
    windowWidth: 'full',
}

it('renders GridRow without crashing', () => {
    shallow(
        <GridRow {...props} />
    )
})

it('renders GridRow correctly', () => {
    let component = renderer.create(
        <GridRow {...props} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    // Make row active
    component = renderer.create(
        <GridRow
            {...props}
            isActive={true}
        />
    )
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    // Make next button active
    component = renderer.create(
        <GridRow
            {...props}
            activeButtonIteration={1}
            isActive={true}
        />
    )
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
