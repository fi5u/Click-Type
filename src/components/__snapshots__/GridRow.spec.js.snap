exports[`test renders GridRow correctly 1`] = `
<div
  className="ui basic buttons GridRow  " />
`;

exports[`test renders GridRow correctly 2`] = `
<div
  className="ui basic buttons GridRow GridRow--is-active " />
`;

exports[`test renders GridRow correctly 3`] = `
<div
  className="ui basic buttons GridRow GridRow--is-active " />
`;
